models/User.js

```diff
@@ -0,0 +1,42 @@
+const mongoose = require("mongoose");
+const { isEmail } = require("validator");
+const bcrypt = require("bcrypt");
+
+const userSchema = new mongoose.Schema({
+  email: {
+    type: String,
+    required: [true, "Please enter an email"],
+    unique: true,
+    lowercase: true,
+    validate: [isEmail, "Please enter a valid email"],
+  },
+  password: {
+    type: String,
+    required: [true, "Please enter a password"],
+    minlength: [6, "Minimum password length is 6 characters"],
+  },
+});
+
+// fire a function before doc saved to db
+userSchema.pre("save", async function (next) {
+  const salt = await bcrypt.genSalt();
```

Comment:
To confirm. Is this salt the same every time? Even if we restart the server or move it to some other machine?
Usually functions like these depend on some seed, and it would cause problems if we try to migrate from one server to another keeping our database as is